crates/red_knot_project/src/metadata/options.rs

```diff
@@ -149,10 +169,10 @@ impl Options {
 #[serde(rename_all = "kebab-case", deny_unknown_fields)]
 pub struct EnvironmentOptions {
     #[serde(skip_serializing_if = "Option::is_none")]
-    pub python_version: Option<PythonVersion>,
+    pub python_version: Option<RangedValue<PythonVersion>>,
```

Comment:
For now, it would be sufficient to only make the key in the `Rules` table ranged because we don't use the range or source for any other value. I instead opted to make all values `Ranged` because the overhead should be small and having the range will make it easier to write good diagnostics.